internal/datanode/flush_manager.go

```diff
@@ -441,7 +455,7 @@ func (m *rendezvousFlushManager) flushDelData(data *DelDataBuf, segmentID Unique
 func (m *rendezvousFlushManager) injectFlush(injection *taskInjection, segments ...UniqueID) {
 	go injection.waitForInjected()
```

Comment:
I'm wondering do we need to use such compliated way for synchronization.
Though the logic looks good me,the following logic seems to work and compact go routine block by flush looks ok to me
	for _, segmentID := range segments {
		m.getFlushQueue(segmentID).inject(injection)
	}